### ISSUE TITLE
Feature request: drawing a single post.

### ISSUE THREAD

**Issue by tofudd** [rr-/szurubooru#198]
Like this: [http://testbooru.donmai.us/posts/random](http://testbooru.donmai.us/posts/random)

**Comment by noirscape** [comment]
(proper title should be "retrieve a random post", drawing implies manually drawing something using the mouse, your context indicates that it is the latter)

**Comment by rr-** [comment]
You can achieve that with `sort:random`

**Comment by mldr2** [comment]
Yes but this show all posts. I need only one.

**Comment by colin969** [comment]
sort:random seems to be seeded though.

I've implemented it in my fork but it'd require more than just calling sort:random to actually pull up a random post sadly.

**Comment by rr-** [comment]
They're not seeded, just cached on the client side. (Basically the web client caches every `GET` request till the user submits a `PUT` or `POST` request.)

**Comment by MystPhysX** [comment]
I'm seeing similar results. 

I was using the API at first with node-fetch (which does not have a cache) and sort:random was giving me the same order each time. I decided to test on the website search and sort:random loads the same order on every browser (mobile phone, desktop, desktop private window) till a `PUT` or `POST` is made like you suggested. Upon refresh after a request the sort order changes but each device still shows the same order.

Is there any way to change this behaviour?

Here's a short video of what I mean showing a normal and private browsing window. https://img.mystphy.sx/ss%20%282019-03-16%20at%2011.29.34%29.mp4

The site is https://nekobooru.xyz

**Comment by rr-** [comment]
Sorry, it's actually on the server side. The behavior of caching `sort:random` is intended, otherwise pagination makes no sense. If you need more sophisticated cache control, amend this file: https://github.com/rr-/szurubooru/blob/master/server/szurubooru/middleware/cache_purger.py

**Comment by MystPhysX** [comment]
So I amended that file and added a simple check to see if 'random' is in the query and purging the cache if so.
The code that does this is
```
if ctx.method != 'GET':
        cache.purge()
elif ctx.has_param('query'):
    if 'random' in ctx.get_param_as_string('query'):
        cache.purge()
```
and it works as intended on the website search when searching for just `sort:random` or even `<tag> sort:random`

However, via the API `https://<booru>/api/posts/?query=sort:random` is giving me a random order but when adding a tag `http://<booru>/api/posts/?query=tag:<tag>,sort:random` is giving me the same post descending order each time. I decided to test with another `sort` token and none of them work (e.g `sort:file-size` doesn't give the right order). I'm guessing the API isn't handling the sort tokens properly?
